Fix profiler throwing javascript errors on IE8 & IE9 #13447

wvdweij opened the pull request.

Commit:
Fix profiler throw javascript errors on IE8 & IE9 #13447

nicolas-grekas commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig:
```diff
@@ -189,17 +189,32 @@
-                    }, false);
+                    /* check for older browsers */
+                    if (!window.addEventListener) {
+                        this.attachEvent("readystatechange", function () {
```
It's onreadystatechange if I'm not wrong?
I'd also deduplicate the closure by putting it in a variable

wvdweij replied:
Thx for feedback, will fix it!

WedgeSama replied:
It can be nice to add a comment like that to help people to handerstand this part of code.
``` js
// Fix ie8&9, does not have XMLHttpRequest.addEventListener method.
// @see https://developer.mozilla.org/fr/docs/Web/API/Element.addEventListener#Internet_Explorer
```

Commit:
Fix profiler throw javascript errors on IE8 & IE9 #13447

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

stof commented on src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig:
```diff
@@ -199,7 +199,14 @@
+                    };
+
+                    /* check for older browsers */
+                    if (!window.addEventListener) {
```
you should use ``this.addEventListener`` here as this is what you use just after

wvdweij replied:
Thx for feedback, changed it

Commit:
Fix profiler throw javascript errors on IE8 & IE9 #13447

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

WedgeSama commented:
There is also 2 addEventListener call in `Collection:form.html.twig` at line 284 & 383.

fabpot commented:
Closing in favor of #13636
